infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -77,12 +77,11 @@ private void processMsg(LogReplicationEntryMsg txMessage) {
             long topologyConfigId = txMessage.getMetadata().getTopologyConfigID();
             long baseSnapshotTs = txMessage.getMetadata().getSnapshotTimestamp();
             long entryTs = txMessage.getMetadata().getTimestamp();
-            long prevTs = txMessage.getMetadata().getPreviousTimestamp();
 
             lastMsgTs = Math.max(persistedLogTs, lastMsgTs);
 
             if (topologyConfigId != persistedTopologyConfigId || baseSnapshotTs != persistedSnapshotStart ||
-                    baseSnapshotTs != persistedSnapshotDone || prevTs != persistedLogTs) {
```

Comment:
removing this means we may process out-of-order messages.  Is that right?

Reply:
Not exactly, the method that eventually calls this class filters out msgs which have lastMsgSentByActive <= lastProcessedTs and currentMsg.timestamp > lastProcessedTs.

SinkBufferManager.java:
> if (preTs <= lastProcessedSeq && currentTs > lastProcessedSeq) {
> ....
> sinkManager.processMessage(dataMessage);
> ...
> }

This leaves out msgs which are expected and would still be inorder because of `preTs <= lastProcessedSeq` . 
In scenarios where the active sends a msg,  which has log entries that have been previously applied by Standby + some new entires, the entries currently would be ignored. And so will all the subsequent msgs.

Reply:
but I believe not every processMessage call checked the same condition, for example, 
SnapshotSinkBufferManager.java:
```
public void processBuffer() {
        while (true) {
            LogReplicationEntryMsg dataMessage = buffer.get(lastProcessedSeq);
            if (dataMessage == null) {
                return;
            }
            sinkManager.processMessage(dataMessage);
            ackCnt++;
            buffer.remove(lastProcessedSeq);
            lastProcessedSeq = getCurrentSeq(dataMessage);
        }
    }
```

would it be possible if we have some out of order messages in buffer?

Reply:
I agree there is scope to clean up/solidify  the checks in these methods.  But I have a some follow-up questions.  Let me get them clarified offline.